target/scripts/helpers/log.sh

```diff
@@ -67,35 +65,37 @@ function _log() {
   case "${1}" in
     ( 'trace' )
       [[ ${LEVEL_AS_INT} -ge 5 ]] || return 0
-      MESSAGE+="  ${CYAN}TRACE  "
+      LEVEL_STRING_WITH_COLOR+="${CYAN}TRACE"
       ;;
 
     ( 'debug' )
       [[ ${LEVEL_AS_INT} -ge 4 ]] || return 0
-      MESSAGE+="  ${PURPLE}DEBUG  "
+      LEVEL_STRING_WITH_COLOR+="${PURPLE}DEBUG"
       ;;
 
     ( 'info' )
       [[ ${LEVEL_AS_INT} -ge 3 ]] || return 0
-      MESSAGE+="   ${BLUE}INF   "
+      LEVEL_STRING_WITH_COLOR+="${BLUE}INFO "
       ;;
 
     ( 'warn' )
       [[ ${LEVEL_AS_INT} -ge 2 ]] || return 0
-      MESSAGE+=" ${LYELLOW}WARNING "
+      LEVEL_STRING_WITH_COLOR+="${LYELLOW}WARN "
       ;;
 
     ( 'error' )
       [[ ${LEVEL_AS_INT} -ge 1 ]] || return 0
-      MESSAGE+="  ${LRED}ERROR  " ;;
+      LEVEL_STRING_WITH_COLOR+="${LRED}ERROR"
+      ;;
 
     ( * )
       _log 'error' "Call to '_log' with invalid log level argument '${1}'"
       return 1
       ;;
   esac
 
-  MESSAGE+="${RESET}]  ${2}"
+  LEVEL_STRING_WITH_COLOR+="${RESET}"
+  MESSAGE="$(date +'%Y-%m-%dT%H:%M:%S%:z')  ${LEVEL_STRING_WITH_COLOR}  ${2}"
```

Comment:
Any reason for including date in all logs?

I think we only did this for change detector for the benefit of tests?

Some deployments however likely have logs managed in a way that timestamps are already part of the logger? So this would just double up on the date and increase disk/network usage? 🤷‍♂️ 

If your deployment logs don't handle this but you require all logs to have a timestamp this would be something Vector should be good for via configuration. There is progress on that front with the syslog sink PR now out of WIP status and waiting on review. Vector might be viable for DMS v14 🙏

Reply:
You've also changed the logging format here to swap log level and date, which does make sense, but if anyone is parsing these logs they'll need to update their logic/regex.

---

Oh and you changed the date format too 😝 

ISO8601 / RFC3339 (more constrained variant) is good. Syslog v1 format also uses RFC3339 format for the timestamp but with further constraints IIRC.

Still, this is probably something better deferred/configured by proper log management. If we're planning to adopt Vector, this change would be short-lived and perhaps introduce an inconvenience for users to adapt to briefly? (_assuming a release with Vector added isn't that far off after a release with this change_)

Reply:
> Oh and you changed the date format too 😝

I noticed the same. At the moment, there are already two log formats used:

```
2023-11-21 17:49:55,021 INFO spawned: 'amavis' with pid 2729
2023-11-21 17:49:55,021 INFO success: amavis entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
[   INF   ]  mail.example.test is up and running
Nov 21 17:49:55 mail amavis[2729]: No ext program for   .zoo, tried: zoo
Nov 21 17:49:55 mail amavis[2729]: No ext program for   .doc, tried: ripole
```

`2023-11-21 17:49:55,021` is from supervisord and `Nov 21 17:49:55` is from `mail.log`.

There shouldn't be introduced a third variant. I would go with the `mail.log` format.

Reply:
`mail.log` is using BSD-syslog IIRC (RFC 3164) and is obsoleted by (RFC 5424) that standardizes syslog format including the timestamp (_to RFC3339 with additional constraints_).

Not sure about `supervisord` format.

Once Vector is added into DMS, users (or test suite) can configure logging much more easily to their needs. So if the motivation for this change was related to an actual need I'd suggest deferring until the switch to Vector is ready.